analyzer/libretime_analyzer/pipeline/organise_file.py

```diff
@@ -1,119 +1,56 @@
+    - The first filename is preserved.
+    - The next filenames receive an UUID append to the name.
     """
-    if not isinstance(audio_file_path, str):
-        raise TypeError(
-            "audio_file_path must be string. Was of type "
-            + type(audio_file_path).__name__
-        )
-    if not isinstance(import_directory, str):
-        raise TypeError(
-            "import_directory must be string. Was of type "
-            + type(import_directory).__name__
-        )
-    if not isinstance(original_filename, str):
-        raise TypeError(
-            "original_filename must be string. Was of type "
-            + type(original_filename).__name__
-        )
-    if not isinstance(metadata, dict):
-        raise TypeError(
-            "metadata must be a dict. Was of type " + type(metadata).__name__
-        )
-    if not os.path.exists(audio_file_path):
-        raise FileNotFoundError(f"audio file not found: {audio_file_path}")
+    filepath = Path(filepath_)
 
-    # Import the file over to it's final location.
-    # TODO: Also, handle the case where the move fails and write some code
-    # to possibly move the file to problem_files.
+    orig_filename = Path(original_filename)
+    dest_path = Path(storage_url)
 
-    max_dir_len = 48
-    max_file_len = 48
-    final_file_path = import_directory
-    orig_file_basename, orig_file_extension = os.path.splitext(original_filename)
+    # Building import path
     if "artist_name" in metadata:
-        final_file_path += (
-            "/" + metadata["artist_name"][0:max_dir_len]
-        )  # truncating with array slicing
-    if "album_title" in metadata:
-        final_file_path += "/" + metadata["album_title"][0:max_dir_len]
-    # Note that orig_file_extension includes the "." already
-    final_file_path += "/" + orig_file_basename[0:max_file_len] + orig_file_extension
+        dest_path /= metadata["artist_name"][0:MAX_DIR_LEN]
 
-    # Ensure any redundant slashes are stripped
-    final_file_path = os.path.normpath(final_file_path)
+    if "album_title" in metadata:
+        dest_path /= metadata["album_title"][0:MAX_DIR_LEN]
 
-    # If a file with the same name already exists in the "import" directory, then
-    # we add a unique string to the end of this one. We never overwrite a file on import
-    # because if we did that, it would mean Airtime's database would have
-    # the wrong information for the file we just overwrote (eg. the song length would be wrong!)
-    # If the final file path is the same as the file we've been told to import (which
-    # you often do when you're debugging), then don't move the file at all.
+    dest_path /= orig_filename.stem[0:MAX_FILE_LEN] + orig_filename.suffix
 
-    if os.path.exists(final_file_path):
-        if os.path.samefile(audio_file_path, final_file_path):
-            metadata["full_path"] = final_file_path
+    # Handle when a file already exists
+    while dest_path.is_file():
```

Comment:
I don't think we will have to chance to generate a duplicate uuid4, but we never know !